### PR TITLE
2.x uploadedFile validation (backported from #4524)

### PR DESCRIPTION
**Pull request by vanquang9387**

### CODE REVIEW

**Comment by dereuromark** [comment]
Looks like you formatted by accident the Validation class file with spaces instead of keeping tabs.

**Comment by vanquang9387** [comment]
Oh right. My IDE didn't show white space diff. I'm fixing it.
Thanks @dereuromark

**Comment by vanquang9387** [comment]
Code has been reformatted.
P.S: Can this PR be merged into 2.7 branch also?

**Comment by vanquang9387** [comment]
@dereuromark @markstory 
Would someone please help me with travis check? Have no idea why check failed.

**Comment by markstory** [comment]
Looks like the tests are [failing because you added a validation rule](https://travis-ci.org/cakephp/cakephp/jobs/86145771). I think you can safely ignore those tests, and I can fix those up post merge.

**Comment by vanquang9387** [comment]
Thanks @markstory :smiley:

File: lib/Cake/Test/Case/Utility/ValidationTest.php [hunk 1]
```diff
@@ -2474,4 +2480,97 @@ public function testFileSize() {
...
+			'size' => 201
+		);
+		$options = array(
+			'minSize' => 500
+		);
+		$this->assertFalse(Validation::uploadedFile($file, $options), 'Too small');
+		$options = array(
+			'maxSize' => 100
+		);
+		$this->assertFalse(Validation::uploadedFile($file, $options), 'Too big');
```
Comment (markstory) [inline lib/Cake/Test/Case/Utility/ValidationTest.php]: There should probably be a test where the rule passes too.

Comment (vanquang9387) [inline lib/Cake/Test/Case/Utility/ValidationTest.php]: Test added.

**Comment by markstory** [comment]
Cleaned up the PHPCS errors and tests. Merged in a99766c8973bc641c934eefa000df5017704afcf.

**Comment by vanquang9387** [comment]
Thanks!

---
Best Regards,

Ngô Văn Quang
Web Division
Rikkeisoft Co,. Ltd

Mobile: +84-94-575-9387
Skype: vanquang9387
Email: quangnv@rikkeisoft.com
http://rikkeisoft.com

**Pull request merged**